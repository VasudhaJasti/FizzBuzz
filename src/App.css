.container{
  width: 80%;
  display: flex;
  align-self: center;
  margin: 50px auto;
  justify-content: space-evenly;
}

.container input{
  padding: 10px 10px;
  margin-right: 10px;
}

.container button{
  padding: 10px 10px;
  background-color: green;
  outline: none;
  border: none;
  color: #fff;
  border-radius: 5;
}