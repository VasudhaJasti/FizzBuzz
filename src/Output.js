import React from "react";

const Output = ({ inputArr }) => {  //Input Array prop from App.js
  const outputArr = []; //Initialization of output array
  inputArr.forEach((val) => {   //For loop for input array based on conditions and push to the output array
    if (isNaN(val) || !val) {   //if value is empty or it is not a number
      outputArr.push("Invalid Item");   
    } else if (val % 15 === 0) {    //If divided by both 3 and 5
      outputArr.push("FizzBuzz");
    } else if (val % 5 === 0) {     //If divided by 5
      outputArr.push("Buzz");
    } else if (val % 3 === 0) {     //If divided by 3
      outputArr.push("Fizz");
    } else {                        // If none of the conditions satisfy, means it is a number which is not divided either by 3 or 5
      outputArr.push(`Divided ${val} by 3`);    
      outputArr.push(`Divided ${val} by 5`);
    }
  });
  return (
    <div className="output-wrapper">
      <h2>Output</h2>
      <div>
          {/* looping of iutput message strings */}
        {outputArr.map((val, i) => (
          <div key={i.toString()} className="list-item">{val}</div>
        ))}
      </div>
    </div>
  );
};
export default Output;
