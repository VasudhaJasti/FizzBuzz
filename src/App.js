import React, { useState } from "react";
import "./App.css";
import Output from "./Output";

class App extends React.Component {
  state = {
    inputArr: [], //stores all input arrays given by user
    inputTxt: "", //stores input value temperorily before add
  };

  onAddToArray = () => {
    //Adding input text to array
    this.setState((prevState) => ({
      inputArr: [...prevState.inputArr, prevState.inputTxt],
      inputTxt: "", //Empty the input after add the value into array
    }));
  };

  render() {
    const {inputArr, inputTxt} = this.state;
    return (
      <div className="container">
        <div className="input-wrapper">
          <h2>Enter Input here...</h2>
          {/* Visual Representaion of input array for user */}
          <p className="visual-array">
            [{inputArr.length > 0 ? inputArr.join(", ") : "Fill the Array"}]
          </p>
          {/* a form to handle input data which prevents default submit */}
          <form onSubmit={(e) => e.preventDefault()}>
            <div className="input-box">
              {/* input elemrnt for adding array element */}
              <input
                onChange={(e) => this.setState({ inputTxt: e.target.value })}
                placeholder="Enter Array element one by one"
                value={inputTxt}
              />
              {/* button for adding input text to array */}
              <button onClick={this.onAddToArray}>Add value to Array</button>
            </div>
          </form>
        </div>
        {/* output and conditional strings for input will be in Output Component */}
        <Output inputArr={inputArr} />
      </div>
    );
  }
}

export default App;
