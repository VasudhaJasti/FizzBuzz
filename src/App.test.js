import React from 'react';
import { shallow } from './enzyme';
import App from './App';
import Output from './Output';

describe('Our test suite', () => {

    it('renders all the mocked animal options', () => {
        const wrapper = shallow(<App/>);
        expect(wrapper).toBeDefined();
        expect(wrapper).toMatchSnapshot();
    });

    it('adding will add input array elements and output messages', () => {
      const wrapper = shallow(<App/>);
      const outputWrapper = wrapper.find(Output)
      expect(outputWrapper.props().inputArr.length).toEqual(0);
      wrapper.find('input').at(0).simulate('change', {target: {value: 6}});
      wrapper.find('button').at(0).simulate('click');
      expect(wrapper.state().inputArr.length).toEqual(1);

      wrapper.setState({
        inputArr: [5,8, '', 15, 9]
      })
      expect(wrapper.find(Output).dive().find('.list-item').length).toEqual(6);
      expect(wrapper.find(Output).dive().find('.list-item').at(0).text()).toEqual('Buzz');
      expect(wrapper.find(Output).dive().find('.list-item').at(1).text()).toEqual('Divided 8 by 3');
      expect(wrapper.find(Output).dive().find('.list-item').at(2).text()).toEqual('Divided 8 by 5');
      expect(wrapper.find(Output).dive().find('.list-item').at(3).text()).toEqual('Invalid Item');
      expect(wrapper.find(Output).dive().find('.list-item').at(4).text()).toEqual('FizzBuzz');
      expect(wrapper.find(Output).dive().find('.list-item').at(5).text()).toEqual('Fizz');
    })

});
