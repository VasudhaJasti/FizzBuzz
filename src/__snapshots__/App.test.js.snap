// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Our test suite renders all the mocked animal options 1`] = `ShallowWrapper {}`;
